cmd/cli/create/create.go

```diff
@@ -78,7 +79,7 @@ func NewCmd() *cobra.Command {
 	return createCmd
 }
 
-func create(cmd *cobra.Command, cmdArgs []string, OC *CreateOptions) error { //nolint:funlen,gocyclo
+func create(cmd *cobra.Command, cfg config.Context, cmdArgs []string, OC *CreateOptions) error { //nolint:funlen,gocyclo
 	ctx := cmd.Context()
```

Comment:
If we pass the client instead that will mean the repo needs to be created in the root command which is something I feel we should avoid.
However I do agree that we should be passing a client to methods called below this instead of the whole config. Opted not to do that here as it would balloon the size of this change, which is already quite large.

Reply:
No we don't have to create the repo in root. Check my comment here on how we can do this https://github.com/bacalhau-project/bacalhau/pull/3959#discussion_r1596903257

Reply:
If we don't create or open the repo in the root command then:
- how will values be read from the config file if one is present?
- how will the clientID be used to sign requests?

Reply:
What I meant is the repo will be created in the PersistentPreRunE in the root, and a client will be created only for the commands that needs one. There shouldn't be any difference on when the repo will be create compared to your current implementation

Reply:
> No we don't have to create the repo in root
>> What I meant is the repo will be created in the PersistentPreRunE in the root

Sorry, but which is it?